Code/GraphMol/RGroupDecomposition/RGroupUtils.cpp

```diff
@@ -208,4 +208,35 @@ std::string toJSON(const RGroupColumns &cols, const std::string &prefix) {
   return res;
 }
 
+void relabelMappedDummies(ROMol &mol, RGroupLabelling inputLabels,
+                          RGroupLabelling outputLabels) {
+  for (auto &atom : mol.atoms()) {
+    if (atom->getAtomicNum()) {
+      continue;
+    }
+    unsigned int atomMapNum = 0;
+    if (inputLabels & AtomMap) {
+      atomMapNum = static_cast<unsigned int>(abs(atom->getAtomMapNum()));
+    }
+    if (!atomMapNum && (inputLabels & Isotope)) {
+      atomMapNum = atom->getIsotope();
+    }
+    if (!atomMapNum && (inputLabels & MDLRGroup)) {
+      atom->getPropIfPresent(common_properties::_MolFileRLabel, atomMapNum);
+    }
+    if (!atomMapNum) {
+      continue;
+    }
+    auto rLabel = "R" + std::to_string(atomMapNum);
+    atom->setProp(common_properties::dummyLabel, rLabel);
+    atom->setAtomMapNum((outputLabels & AtomMap) ? atomMapNum : 0);
+    atom->setIsotope((outputLabels & Isotope) ? atomMapNum : 0);
```

Comment:
I think this is going too far with keeping the code short (we don't have to pay by the line of code ;-) ), how about this:
```suggestion
    if(outputLabels & AtomMap) {
      atom->setAtomMapNum(atomMapNum);
    } else {
      atom->setAtomMapNum(0);
    }
    if(outputLabels & Isotope) {
      atom->setIsotope(atomMapNum);
    } else {
      atom->setIsotope(0);
    }
```

Reply:
This is part of my "Greg alertness test" campaign: the day you won't flag a ternary conditional operator or >5-line-long logic test I will start worrying. Great that this has not happened yet.
(I still prefer the ternary operator to the above, but that does not matter as much)